### ISSUE TITLE
there is no check if file was downloaded correctly

### ISSUE THREAD

**Issue by vogel** [stellar/stellar-core#1128]
With current code following scenerio is possible:

1. get-remote-file can fail but still create .gz file

Invalid file (or no file at all) is downloaded from server with curl/wget -o command by GetRemoteFileWork. In acceptance tests wget is used. You can see that an output file is always created as a result of `wget` command:

    wget -q http://example.abc/file -O file.gz
    ls file.gz
    -rw-rw-r-- 1 rmalinowski rmalinowski     0 sty 16 14:46 file.gz

An empty file was created, but it is not valid.

2. get-remote-file will be retried, but in the result it can still leave invalid file behind (if the server fails)
3. get-remote-file failure will made batch-download-ledger retry
4. batch-download-ledger retry will now call gunzip-file on empty file, that will fail each time
5. batch-download-ledger will be retried again
6. after some time catchup-minimal will fail and will be retried

and so on...

Solution no. 1: add a temporary step.
For example:

rm file.gz.tmp
wget http://history/file -o file.gz.tmp
if succeeded
mv file.gz.tmp file.gz
if succeeded
gunzip file.gz

It fixes half the problem. There still can be issue where wget/curl reports success, but file was corrupted in some way. In that case we need:

Solution no. 2: add a way to remove .gz file if gunzip failed, then retry download.

Currently it does not seem possible.

**Comment by vogel** [comment]
It does not seem like caused by https://github.com/stellar/stellar-core/pull/1074/commits/f4dbf728479155ed545bcfe36dac8f40f12c6d12 because that only adds additional gunzip and does not remove any re-downloading feature (that was not there).

Still strange why it shows on jenkins and works properly on my local PC. I'll look into code from Mar 15 to see what have changed.

**Comment by mvaneijk** [comment]
I suspected that my fix (of f4dbf72) only solved the symptom. The reasoning was that the download is done as a child worker process of the unzip worker process and therefor should succeed before the unzip can take place.  
According to that reasoning, the file cannot be unzipped before the download has succeeded because the file does not exist. However, the file did exist in my situation, and I added a solution for it.

Maybe inserting the child work in the map<> is not correct sometimes?
